src/obu.rs

```diff
@@ -2098,40 +2098,13 @@ fn parse_tile_hdr(tiling: &Rav1dFrameHeader_tiling, gb: &mut GetBits) -> Rav1dTi
     }
 }
 
-/// Check that we haven't read more than `obu_len`` bytes
-/// from the buffer since `init_bit_pos`.
-fn check_for_overrun(
-    c: &mut Rav1dContext,
-    gb: &mut GetBits,
-    init_bit_pos: usize,
-    obu_len: usize,
-) -> c_int {
-    // Make sure we haven't actually read past the end of the `gb` buffer
-    if gb.has_error() != 0 {
-        writeln!(c.logger, "Overrun in OBU bit buffer");
-        return 1;
-    }
-
-    let pos = gb.pos();
-
-    // We assume that `init_bit_pos` was the bit position of the buffer
-    // at some point in the past, so cannot be smaller than `pos`.
-    assert!(init_bit_pos <= pos);
-
-    if pos - init_bit_pos > 8 * obu_len {
-        writeln!(c.logger, "Overrun in OBU bit buffer into next OBU");
-        return 1;
-    }
-
-    0
-}
-
 unsafe fn parse_obus(
     c: &mut Rav1dContext,
     r#in: &CArc<[u8]>,
     props: &Rav1dDataProps,
+    mut gb: &mut GetBits,
 ) -> Rav1dResult<usize> {
```

Comment:
Now that `gb` is from the caller, we could remove the `usize` from the return, as the caller can just call `gb.len()`.